### PR TITLE
Improvement : Category management drag area

### PR DESCRIPTION
**Pull request by kumarpalsinh25**
fixes, https://github.com/acterglobal/a3-meta/issues/637#issuecomment-2487357906

- Forked [DragAndDropLists](https://github.com/philip-brink/DragAndDropLists) Plugin
- Did required changes in [Forked Repository](https://github.com/kumarpalsinh25/DragAndDropLists/commit/4b086bcb22c41e28dd4e5094585d3d3d1f799977)
- Used forked version in Acter by updating `app/pubspec.yaml` 
- Did required changes in code to better manage drag area

Reference Video:

https://github.com/user-attachments/assets/8353b265-bc05-4582-a652-bf5673735f48

### CODE REVIEW

**Review by gnunicorn — APPROVED** [review]
looks fine but needs minor fixes

File: app/lib/features/categories/utils/category_utils.dart [hunk 1]
```diff
@@ -53,7 +53,7 @@ class CategoryUtils {
     //ADD UN-CATEGORIES ITEM to LAST POSITION
     CategoryModelLocal unCategorized = CategoryModelLocal(
       entries: unCategoriesEntriesList,
-      title: '',
+      title: 'Un-categorized',
```
[the remark this answers is not in the document]
Comment (kumarpalsinh25) [reply app/lib/features/categories/utils/category_utils.dart:56]: Getting difficulty in getting context for L10n as this function is called from Providers and Multiple test-cases overrides.

[the remark this answers is not in the document]
Comment (gnunicorn) [reply app/lib/features/categories/utils/category_utils.dart:56]: Shouldn't the model then support having `String? title` and it being set to the l10n if not set at time of render? Hard-coding english strings that are visible to the user are not the way though...

[the remark this answers is not in the document]
Comment (kumarpalsinh25) [reply app/lib/features/categories/utils/category_utils.dart]: I tried `String? title` but then realised that all the places from where `CategoryUtils` method used are from providers. 

Look like PR is merged but if you suggest then I can rollback entire title itself as it doesn't mandatory in this use-case.

**Pull request merged**